### PR TITLE
enhancement: polish send form for stardust

### PR DESCRIPTION
**Pull request by nicole-obrien**
## Summary
This PR improves the send form so that the units and amount work independent of the the token type. We will still need a task to refactor the units and formatters into an assets module

### Changelog
```
- Rename formatTokenAmount functions
- Replace unit enum with local one just for the metric prefix
- Add logic for available amount button in AssetAmountInput
- Make unit dropdown based on tokenmetadata
- Type recipient object
```

## Relevant Issues
Please list any related issues (e.g. bug, task).

## Type of Change
Please select any type below that is applicable to your changes, and delete those that are not.
- [ ] __Breaking__ - any change that would cause existing functionality to not work as expected
- [ ] __Chore__ - refactoring, build scripts or anything else that isn't user-facing
- [ ] __Docs__ - changes to the documentation
- [ ] __Fix__ - a change which fixes an issue
- [ ] __New__ - a change which implements a new feature
- [x] __Update__ - a change which updates existing functionality

### Platforms
Please select any platforms where your changes have been tested.
- __Desktop__
	- [x] MacOS
	- [ ] Linux
	- [ ] Windows
- __Mobile__
	- [ ] iOS
	- [ ] Android

### Instructions
Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

## Checklist
Please tick all of the following boxes that are relevant to your changes, and delete those that are not.
- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/components/inputs/AssetAmountInput.svelte [hunk 1]
```diff
@@ -1,53 +1,61 @@
...
+    $: if (!asset?.metadata.useMetricPrefix) {
+        if (unit === asset?.metadata.unit) {
+            rawAmount = Number(amount) * 10 ** asset?.metadata.decimals
+        } else if (unit === asset?.metadata.subunit) {
+            rawAmount = Number(amount)
+        }
+    } else if (asset?.metadata.useMetricPrefix) {
+        rawAmount = Number(amount) * UNIT_MAP?.[unit?.substring(0, 1)] ?? 0
+    }
+
```
Comment (maxwellmattryan) [inline packages/shared/components/inputs/AssetAmountInput.svelte:31]: We could probably make this into a function like `generateRawAmount(amount: number, tokenMetadata: ITokenMetadata): number`, might be handy to have later on.

File: packages/shared/components/inputs/AssetAmountInput.svelte [hunk 2]
```diff
@@ -1,53 +1,61 @@
...
+        if (!asset?.metadata.useMetricPrefix) {
+            const balance = (asset?.balance?.available ?? 0) / 10 ** asset?.metadata.decimals
+            amount = balance.toString()
+            unit = asset.metadata.unit
+        } else if (asset?.metadata.useMetricPrefix) {
+            const metricUnit = getUnit(asset?.balance?.available ?? 0)
+            const balance = (asset?.balance?.available ?? 0) / UNIT_MAP[metricUnit].val
+            amount = balance.toString()
+            unit = metricUnit + asset.metadata.unit
+        }
```
Comment (maxwellmattryan) [inline packages/shared/components/inputs/AssetAmountInput.svelte:42]: Same for here; it feels like we could refactor this logic into a function and cleanup this component a bit.

File: packages/shared/lib/core/wallet/utils/formatTokenAmountBestMatch.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { formatNumber } from '@lib/currency'
+import { ITokenMetadata } from '../interfaces'
+import { formatTokenAmountDefault } from './formatTokenAmountDefault'
+import { getUnit, UNIT_MAP } from '@lib/units'
```
Comment (maxwellmattryan) [inline packages/shared/lib/core/wallet/utils/formatTokenAmountBestMatch.ts:4]: Import order should be in order of outermost modules first

File: packages/shared/lib/units.ts [hunk 4]
```diff
@@ -1,20 +1,35 @@
-import { Unit } from '@iota/unit-converter'
 import Big from 'big.js'
+import BigNumber from 'bignumber.js'
 import { getCurrencyPosition, formatNumber } from 'shared/lib/currency'
 
 // Set this to avoid small numbers switching to exponential format
 Big.NE = -20
 
+export enum Unit {
```
Comment (maxwellmattryan) [inline packages/shared/lib/units.ts:8]: Since this is for IOTA, should we name this `IotaUnit` rather than just `Unit`?

**Review by maxwellmattryan — CHANGES_REQUESTED** [review]
Code looks good! Just left some small comments :fire:

File: packages/shared/lib/units.ts (hunk 4 above)
Comment (nicole-obrien) [reply packages/shared/lib/units.ts:8]: It is going to be used for any Metric amount, but its not needed at the moment and the refactor of the units is a can of worms. I will open an issue to refactor the units and formatters into the assets module.

**Review by maxwellmattryan — APPROVED** [review]
LGTM

**Pull request merged**